hmcts/pre-api
# S28 3269: `PUT /user/{id}` Restricted to Admin Users

### JIRA ticket(s)
- S28-3269


### Change description
- Restrict access to `PUT /users/{id}` endpoint to only admin level users (Super User, Level 1)
- Expand functional tests to test as specific roles (code taken from a different pr #653) 




> [!IMPORTANT]
> This PR requires manual testing by QA. Please notify the QA team @hmcts/pre-rec-evidence-qa.

**QA instructions**
- Test: Attempt to create a new user via the `PUT /users/{id}` endpoint whilst acting as a user with in role `Super User` (should be successful- 201 response)
- Test: Attempt to create a new user via the `PUT /users/{id}` endpoint whilst acting as a user with in role `Level 1` (should be successful- 201 response)
- Test: Attempt to create a new user via the `PUT /users/{id}` endpoint whilst acting as a user with in role `Level 2` (should not be successful- 403 response)
- Test: Attempt to create a new user via the `PUT /users/{id}` endpoint whilst acting as a user with in role `Level 3` (should not be successful- 403 response)
- Test: Attempt to create a new user via the `PUT /users/{id}` endpoint whilst acting as a user with in role `Level 4` (should not be successful- 403 response)
